### PR TITLE
Send email notification when user email changes.

### LINKED ISSUE #6125: Send email notifications when email address or passwords are changed
The goal of this ticket is to send an email when there is an email address or password change. Later this email could be extended to other security actions.

Your email address and password are how you safeguard your Piwik account, so monitoring it for any changes is critical. We could send emails when there are security related changes to your account, such as a password change. 

The information would give you more insight into when and where the account change took place, including 
* the date and time and
*  details on the device the device the changes were made on such as
* the browser it was running, 
* the Operating System (OS), 
* IP address, 
* and approximate physical location.

Example at linkedin:

password-change

### LINKED ISSUE #13533: Require email verification when changing email address.
Related to https://github.com/matomo-org/matomo/issues/2932 and https://github.com/matomo-org/matomo/issues/6125 (the latter could maybe also be added in the same change)

At the moment every Matomo user could change their E-Mail address to everything they want without any verification (apart from syntax, see https://github.com/matomo-org/matomo/issues/11796). This allows every Matomo user to send an unlimited amount of (for them) SPAM E-Mails to anyone without them ever opting in to receiving them.

When someone tries to change their email address (after confirming their password; https://github.com/matomo-org/matomo/issues/2932) the change should only be saved if the user was able to confirm an link in a sent mail. 
In addition an email should be sent to the old email address informing them that they are loosing access to the account (and to inform an admin if they don't know about the change)
https://github.com/matomo-org/matomo/issues/6125

### PR DESCRIPTION
**Pull request by diosmosis**
Fixes #6125

emailscreen

Not sure if the email text is ok, let me know if it should change.

### CODE REVIEW

File: plugins/UsersManager/API.php
Comment (sgiehl) [inline plugins/UsersManager/API.php]: Might it maybe make sense to add some more details about the device if available? `getDeviceName` only give `desktop, smartphone, ...` Maybe we could if available also add `getBrand()` and `getModel()`. So we could print something like `smartphone (Sony Xperia Z5)` or only `dekstop` if not specified

**Review by sgiehl — APPROVED** [review]
Works nicely. Maybe @mattab or @tsteur should review the text as well.
Left a small suggestion, but it's not really necessary. Good to merge if text is approved.

**Comment by Findus23** [comment]
:+1: Looking great. 

But I'd prefer to keep #13533 open as this doesn't solve the spam-possibility or the double-opt in feature. (even though I am not sure anymore how important it is)

Instead https://github.com/matomo-org/matomo/issues/6125 can be closed.

File: plugins/UsersManager/API.php
Comment (diosmosis) [reply plugins/UsersManager/API.php]: 👍 will do

**Comment by diosmosis** [comment]
@Findus23 I can add a password changed email.

**Comment by mattab** [comment]
Feedback:
* Looks good!
* To make email more friendly, let's add 
  * at the start "Hello,"
  * at the end:  "Thank you for using Matomo! <br>The Matomo team"  or similar?

> I can add a password changed email

Yes be important to notify on password change too :+1:

**Comment by diosmosis** [comment]
Updated.

**Review by sgiehl — CHANGES_REQUESTED** [review]
left some minor comments. Otherwise looks good to be merged.
Might that be something that needs to be considered for WhiteLabel plugin? ping @tsteur

File: plugins/UsersManager/templates/_emailChangedEmail.twig
Comment (tsteur) [inline plugins/UsersManager/templates/_emailChangedEmail.twig]: is the device name always set? Looking at the code it may be empty but probably isn't really often the case. May not be too useful though to show whether it was updated from desktop or mobile if no device was detected. Not too important though and fine to leave it just like that.

File: plugins/UsersManager/API.php
Comment (tsteur) [inline plugins/UsersManager/API.php]: could we maybe mostly reuse the same code for `sendEmailChangedEmail` and `sendPasswordChangedEmail ` and also it's templates and only have different mail subject and also different leading sentence in the email? not too important though.

File: plugins/UsersManager/templates/_emailChangedEmail.twig
Comment (tsteur) [reply plugins/UsersManager/templates/_emailChangedEmail.twig]: I reckon there can be various cases where user agent may not be set when details are updated through the API. Therefore device name may be empty on ocassion

File: plugins/UsersManager/API.php
Comment (diosmosis) [reply plugins/UsersManager/API.php]: Thought about it, but since there were just two emails, and the code isn't really complicated (just setting things on the view & Mail instance), it didn't seem that important. Maybe if there are more changes to a user that we should notify them about?

**Comment by diosmosis** [comment]
@tsteur did you see @sgiehl's comment about WhiteLabel?

**Comment by tsteur** [comment]
Yes, sorry forgot to reply. I haven't tested it but I think it should be all fine as it also works for all the other mails. Might need to double check at some point though whether we replace the brand eg in all `Mails.send` etc

**Comment by diosmosis** [comment]
Updated.

**Comment by tsteur** [comment]
minor last tweak... could we use `ucfirst` for device name? (desktop)
image

Also I thought it was sending out duplicate emails until I realised it was actually all fine but the source and target email was the same but `source=test+1@domain.com` and `target=test+2@domain.com`. No bug and nothing needs to change. Just in case someone tests it as well and runs in the same issue.

Looks good to merge if tests pass

**Comment by mattab** [comment]
@diosmosis in my tests, when the email address is updated, it seems the email is sent twice, could you double check emails are not sent twice?

**Comment by tsteur** [comment]
See the comments... you aren't change your email to your own email are you?

**Comment by mattab** [comment]
> you aren't change your email to your own email are you?

oops I should have seen this one coming :-D  No more issue.

**Pull request merged**
